empyrical/stats.py

```diff
@@ -140,11 +142,14 @@ def cum_returns(returns, starting_value=0):
     if len(returns) < 1:
         return type(returns)([])
 
-    if np.isnan(np.asanyarray(returns)[0]):
-        returns = returns.copy()
-        returns[0] = 0.
+    if isinstance(returns, pd.DataFrame):
+        df_cum = (returns.fillna(0) + 1).cumprod()
 
-    df_cum = np.exp(nancumsum(np.log1p(returns)))
+    else:
+        if np.all(np.isnan(np.asanyarray(returns)[0])):
+            returns = returns.copy()
+            returns[0] = 0.
+        df_cum = np.exp(nancumsum(np.log1p(returns)))
```

Comment:
Agreed, I was going to suggest this too.  If we can use the same code for 1D and 2D+ arrays, then we don't need to test them both as essentially isolated functions.  I think we need to pass `axis=0` to `nancumsum` here.  We also need to fix the assignment to the zero column on the line above, in the case of a DataFrame.  

Would also be nice if this isn't DataFrame-specific, as opposed to an ndarray with more than 1 dimension.